examples/ota-requestor-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,202 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CHIPDeviceManager.h"
+#include "DeviceCallbacks.h"
+#include "esp_heap_caps_init.h"
+#include "esp_log.h"
+#include "esp_netif.h"
+#include "esp_spi_flash.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "nvs_flash.h"
+#include <app/server/Server.h>
+
+#include <cmath>
+#include <cstdio>
+#include <string>
+#include <vector>
+
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <credentials/examples/DeviceAttestationCredsExample.h>
+
+#include <lib/support/ErrorStr.h>
+
+#include "OTARequesterImpl.h"
+#include <argtable3/argtable3.h>
+#include <esp_console.h>
+
+using namespace ::chip;
+using namespace ::chip::System;
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceManager;
+using namespace ::chip::DeviceLayer;
+
+const char * TAG = "ota-requester-app";
+
+static DeviceCallbacks EchoCallbacks;
+
+struct CmdArgs
+{
+    struct arg_str * ipAddr;
+    struct arg_int * nodeId;
+    struct arg_end * end;
+};
+
+CmdArgs queryImageCmdArgs, applyUpdateCmdArgs;
+
+void QueryImageTimerHandler(Layer * systemLayer, void * appState)
+{
+    ESP_LOGI(TAG, "Calling SendQueryImageCommand()");
+    OTARequesterImpl::GetInstance().SendQueryImageCommand(queryImageCmdArgs.ipAddr->sval[0], queryImageCmdArgs.nodeId->ival[0]);
+}
+
+void ApplyUpdateTimerHandler(Layer * systemLayer, void * appState)
+{
+    ESP_LOGI(TAG, "Calling SendApplyUpdateRequestCommand()");
+    OTARequesterImpl::GetInstance().SendApplyUpdateRequestCommand(queryImageCmdArgs.ipAddr->sval[0],
+                                                                  queryImageCmdArgs.nodeId->ival[0]);
+}
+
+int ESPQueryImageCmdHandler(int argc, char ** argv)
+{
+    int nerrors = arg_parse(argc, argv, (void **) &queryImageCmdArgs);
+    if (nerrors != 0)
+    {
+        arg_print_errors(stderr, queryImageCmdArgs.end, argv[0]);
+        return 1;
+    }
+    ESP_LOGI(TAG, "ipAddr:%s nodeId:%x", queryImageCmdArgs.ipAddr->sval[0], queryImageCmdArgs.nodeId->ival[0]);
+
+    /* Start on shot timer to Query for OTA image once commissinoning is complete */
+    chip::DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(1 * 1000), QueryImageTimerHandler, nullptr);
+    return 0;
+}
+
+int ESPApplyUpdateCmdHandler(int argc, char ** argv)
+{
+    int nerrors = arg_parse(argc, argv, (void **) &applyUpdateCmdArgs);
+    if (nerrors != 0)
+    {
+        arg_print_errors(stderr, applyUpdateCmdArgs.end, argv[0]);
+        return 1;
+    }
+    ESP_LOGI(TAG, "ipAddr:%s nodeId:%x", applyUpdateCmdArgs.ipAddr->sval[0], applyUpdateCmdArgs.nodeId->ival[0]);
+
+    /* Start on shot timer to Query for OTA image once commissinoning is complete */
```

Comment:
Does it make sense to try to apply right after commissioning? Shouldn't it first hear back from QueryImageResponse?